### PR TITLE
User features

### PR DESCRIPTION
**Pull request by emrecetinkayadev**

### CODE REVIEW

File: src/blog/users/service.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def update(user_id: int, user_in: UserUpdate) -> t.Optional[User]:
+    user_in = user_in.dict(exclude_unset=True)
+    for index, user in enumerate(user_table):
+        if user["id"] == user_id:
+            user_in["id"] = user_id
+
+            # Passing new password and delete it.
+            user_in["password"] = hash_password(
```
Comment (eonalion) [inline src/blog/users/service.py]: What if password and new password were not set in `user_in` model? (For example, user updated only username)

File: src/blog/users/views.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    """
+    user = get(user_id=user_id)
+    # Checks if user has post or not.
+    post_count = len(get_posts_by_user_id(user_id=user_id))
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=[{"msg": "User with this ID doesn't exist"}]
+        )
+    elif post_count > 0:
```
Comment (eonalion) [inline src/blog/users/views.py:77]: System should delete posts by itself in a cascade manner if the author is deleted (or at least mark user as inactive and filter out posts by this flag).
Let's skip this for now and add this part when we start support database.

**Review by eonalion — CHANGES_REQUESTED** [review]

File: src/blog/posts/service.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ def get(post_id: int) -> t.Optional[Post]:
     for post in post_table:
         if post["id"] == post_id:
             searched_post = post
-
+    breakpoint()
```
Comment (eonalion) [inline src/blog/posts/service.py]: You need to remove breakpoints before committing the changes.

File: src/blog/posts/views.py [hunk 4]
```diff
@@ -71,9 +65,9 @@ def delete_post(post_id: PrimaryKey):
             detail=[{"msg": "Post with this id doesn't exist"}],
         )
     delete(post_id=post_id)
-        # get post by id using service method
-        # if not post raise 404 HTTPException (check specification: https://fastapi.tiangolo.com/tutorial/handling-errors/ )
-        # else post exists, call delete from service
-        # usually no need to return anything from delete, sometimes the deleted id can be returned,
-        # but it's ok to return just HTTP 200 response
-        # (If there are no exceptions raised and no return, FastAPI automatically return HTTP 200)
+    # get post by id using service method
```
Comment (eonalion) [inline src/blog/posts/views.py]: I think now you can remove all this comments.

File: src/blog/users/service.py (hunk 1 above)
Comment (emrecetinkayadev) [reply src/blog/users/service.py]: It's resolved by moving hashing logic to models.

File: src/blog/users/views.py (hunk 2 above)
Comment (emrecetinkayadev) [reply src/blog/users/views.py:77]: Okey, we will implement it in the future.

File: src/blog/posts/service.py (hunk 3 above)
Comment (emrecetinkayadev) [reply src/blog/posts/service.py]: I forget to delete it. Now it's fixed.

File: src/blog/posts/views.py (hunk 4 above)
Comment (emrecetinkayadev) [reply src/blog/posts/views.py]: All comments are removed.

**Pull request merged**